project/src/main/java/com/google/sps/servlets/CompetitionInfoServlet.java

```diff
@@ -14,25 +14,45 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
+import com.google.sps.data.*;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import com.google.gson.Gson;
-import com.google.sps.data.*;  
 
 @WebServlet("/competitionInfo")
 public class CompetitionInfoServlet extends HttpServlet {
-
+  List<Competition> usersCompetitions = new ArrayList<>();
```

Comment:
This is a class member, what happens when 2 requests come in at the same time, want this variable be mutated by the 2 requests

May be you getUserCompetitions to return a list instead of setting usersCompetitions